flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptivebatch/BatchJobRecoveryTest.java

```diff
@@ -1212,4 +1234,45 @@ public Optional<ResourceID> storesLocalResourcesOn() {
             };
         }
     }
+
+    private void waitUntilAllExecutionsDeployed(
+            JobVertexID vertexId, AdaptiveBatchScheduler scheduler, long maxWaitMillis)
+            throws Exception {
+        checkArgument(maxWaitMillis >= 0);
+
+        // this is a poor implementation - we may want to improve it eventually
```

Comment:
No need to introduce a timeout for this method. Test cases will timeout if not completed in time.